### PR TITLE
roachtest: update pg_regress diff

### LINKED ISSUE #120590: roachtest: pg_regress failed
roachtest.pg_regress [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestWeeklyBazel/14417980?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestWeeklyBazel/14417980?buildTab=artifacts#/pg_regress) on master @ [34caf9d2c267f95f6aef2713708806e6f14948d6](https://github.com/cockroachdb/cockroach/commits/34caf9d2c267f95f6aef2713708806e6f14948d6):

```
  DECLARE
-@@ -5787,13 +6592,13 @@
+@@ -5787,13 +6574,13 @@
      SELECT * INTO result FROM partitioned_table WHERE a = a_val;
      RETURN result;
  END; $$ LANGUAGE plpgsql;
@@ -250238,7 +249969,7 @@
  CREATE OR REPLACE FUNCTION list_partitioned_table()
  RETURNS SETOF partitioned_table.a%TYPE AS $$
  DECLARE
-@@ -5806,14 +6611,13 @@
+@@ -5806,14 +6593,13 @@
      END LOOP;
      RETURN;
  END; $$ LANGUAGE plpgsql;
@@ -250259,19 +249990,25 @@
  --
  -- Check argument name is used instead of $n in error message
  --
-@@ -5822,6 +6626,9 @@
+@@ -5822,6 +6608,15 @@
    GET DIAGNOSTICS x = ROW_COUNT;
    RETURN;
  END; $$ LANGUAGE plpgsql;
 -ERROR:  "x" is not a scalar variable
 -LINE 3:   GET DIAGNOSTICS x = ROW_COUNT;
 -                          ^
-+ERROR:  at or near "return": syntax error: missing expression
-+DETAIL:  source SQL:
-+BEGIN
-+  GET DIAGNOSTICS x = ROW_COUNT;
-+  RETURN;
-+  ^
++ERROR:  unimplemented: attempted to use a PL/pgSQL statement that is not yet supported
++HINT:  You have attempted to use a feature that is not yet implemented.
++
++Please check the public issue tracker to check whether this problem is
++already tracked. If you cannot find it there, please report the error
++with details by creating a new issue.
++
++If you would rather not post publicly, please contact us directly
++using the support form.
++
++We appreciate your feedback.
++
 diff -U3 --label=/mnt/data1/postgres/src/test/regress/expected/hash_part.out --label=/mnt/data1/postgres/src/test/regress/results/hash_part.out /mnt/data1/postgres/src/test/regress/expected/hash_part.out /mnt/data1/postgres/src/test/regress/results/hash_part.out
 --- /mnt/data1/postgres/src/test/regress/expected/hash_part.out
 +++ /mnt/data1/postgres/src/test/regress/results/hash_part.out
If the diff is expected, copy regression.diffs from the test artifacts to pkg/cmd/roachtest/testdata/regression.diffs
test artifacts and logs in: /artifacts/pg_regress/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-14417980/pg-regress/1710573517952/1710574210404)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*pg_regress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

Jira issue: CRDB-36763

### PR DESCRIPTION
**Pull request by yuzefovich**
Fixes: #120590.

Epic: None

Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/testdata/regression.diffs [hunk 1]
```diff
@@ -39233,6 +39233,23 @@ diff -U3 --label=/mnt/data1/postgres/src/test/regress/expected/horology.out --la
  SELECT time with time zone '01:30-08' - interval '02:01' AS "23:29:00-08";
   23:29:00-08 
  -------------
+@@ -1034,14 +696,14 @@
+             + interval '02:01' AS time with time zone) AS time) AS "07:31:00";
+  07:31:00 
+ ----------
+- 07:31:00
++ 08:31:00
```
Comment (michae2) [inline pkg/cmd/roachtest/testdata/regression.diffs:39241]: What is causing this difference?

Comment (yuzefovich) [reply pkg/cmd/roachtest/testdata/regression.diffs]: I wish I knew - I ran the roachtest manually from the gceworker, and I am getting this spurious diffs occasionally. Reverted.

**Review by michae2 — APPROVED** [review]
LGTM!

**Comment by yuzefovich** [comment]
Thanks for the careful review!

bors r+

**Pull request merged**